exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,179 @@
+			player.takePlayerOutOfPenaltyBox();
+			isOutOfPenaltyBox = true;
+
+			updatePlayerLocation(result);
+		} else {
+			System.out.println(playersInGame.get(playerLocation) + " is not getting out of the penalty box");
+			isOutOfPenaltyBox = false;
+			}
+	}
+
+	private void updatePlayerLocation(final int roll) {
+		playerOrder[playerLocation] += roll;
+		if (playerOrder[playerLocation] > LAST_POSITION) {
+			playerOrder[playerLocation] -= RESET_POSITION;
+		}
+
+		System.out.println(playersInGame.get(playerLocation)
+				+ "'s new location is "
+				+ playerOrder[playerLocation]);
+		System.out.println("The category is " + getCategory());
+		askQuestion();
+	}
+
+	private void askQuestion() {
+		if (getCategory().equals(CATEGORY_POP))
+			System.out.println(popQuestionsList.removeFirst());
+		if (getCategory().equals(CATEGORY_SCIENCE))
+			System.out.println(scienceQuestionsList.removeFirst());
+		if (getCategory().equals(CATEGORY_SPORTS))
+			System.out.println(sportsQuestionsList.removeFirst());
+		if (getCategory().equals(CATEGORY_ROCK))
+			System.out.println(rockQuestionsList.removeFirst());
+	}
+	
+	
+	private String getCategory() {
+    	String[] boardSpaces = {
+    			CATEGORY_POP, CATEGORY_SCIENCE, CATEGORY_SPORTS, CATEGORY_ROCK,
+				CATEGORY_POP, CATEGORY_SCIENCE, CATEGORY_SPORTS, CATEGORY_ROCK,
+				CATEGORY_POP, CATEGORY_SCIENCE, CATEGORY_SPORTS, CATEGORY_ROCK};
+    	return boardSpaces[playerOrder[playerLocation]];
+	}
+
+	public boolean wasAnsweredCorrectly() {
+		if (playersInPenaltyBox[playerLocation]){
+			if (isOutOfPenaltyBox) {
+				System.out.println("Answer was correct!!!!");
+				return calculateScoreAndDisplay();
+			} else {
+				playerLocation++;
+				resetPlayerLocationToZero();
+				return true;
+			}
+		} else {
+			// Don't touch this spelling mistake upon pain of death
+			// It seems to hold the whole universe together
+			System.out.println("Answer was corrent!!!!");
+			return calculateScoreAndDisplay();
+		}
+	}
+
+	public boolean wasAnsweredIncorrectly(){
+		Player player = playersInGame.get(playerLocation);
+		System.out.println("Question was incorrectly answered");
+		System.out.println(playersInGame.get(playerLocation)+ " was sent to the penalty box");
+
+		player.putPlayerInPenaltyBox();
+		//playersInPenaltyBox[playerLocation] = true;
+
+		playerLocation++;
+		resetPlayerLocationToZero();
+		return true;
+	}
+
+	private boolean calculateScoreAndDisplay() {
+		playerPurses[playerLocation]++;
+		System.out.println(playersInGame.get(playerLocation)
+				+ " now has "
+				+ playerPurses[playerLocation]
+				+ " Gold Coins.");
+
+		return determineWinner();
+	}
+
+	private boolean determineWinner() {
+		boolean winner = didPlayerWin();
+		playerLocation++;
+		resetPlayerLocationToZero();
+		return winner;
+	}
+
+	private void resetPlayerLocationToZero() {
+		if (playerLocation == playersInGame.size()) playerLocation = 0;
+	}
+
+	private boolean didPlayerWin() {
```

Comment:
This method name might make more sense if it read "playerDoesNotHaveSixCoins" as I am not sure it tells you if they won or not